### PR TITLE
RPC endpoint adaptation

### PR DESCRIPTION
**Pull request by YohanTz**
### Commits from starknet-specs:
- [x]  event filter support for disjunction over values in the same key: [[#55](https://github.com/starkware-libs/starknet-specs/pull/55)](https://github.com/starkware-libs/starknet-specs/pull/55)
- [x]  starkNet 0.11.0 adjustments (new class structure, declare v2): [[#65](https://github.com/starkware-libs/starknet-specs/pull/65)](https://github.com/starkware-libs/starknet-specs/pull/65)
- [x]  add "required" to schemas with required properties: [[#67](https://github.com/starkware-libs/starknet-specs/pull/67)](https://github.com/starkware-libs/starknet-specs/pull/67)
- [x]  make transaction type explicit: [[#68](https://github.com/starkware-libs/starknet-specs/pull/68)](https://github.com/starkware-libs/starknet-specs/pull/68)
- [ ]  add simulate_tx endpoint and refactor trace: [[#69](https://github.com/starkware-libs/starknet-specs/pull/69)](https://github.com/starkware-libs/starknet-specs/pull/69)
- [x]  add EVENTS_CHUNK: [[#70](https://github.com/starkware-libs/starknet-specs/pull/70)](https://github.com/starkware-libs/starknet-specs/pull/70)
- [x]  ~~felt encoding change - no leading zeros: [[#72](https://github.com/starkware-libs/starknet-specs/pull/72)](https://github.com/starkware-libs/starknet-specs/pull/72)~~ (nothing to change)
- [x]  state update response split to pending/none pending: [[#74](https://github.com/starkware-libs/starknet-specs/pull/74)](https://github.com/starkware-libs/starknet-specs/pull/74)
- [x]  state_update 0.11.0 refactor: [[#76](https://github.com/starkware-libs/starknet-specs/pull/76)](https://github.com/starkware-libs/starknet-specs/pull/76)
- [x]  bulk API, change estimateFee and simulate to take arrays: [[#77](https://github.com/starkware-libs/starknet-specs/pull/77)](https://github.com/starkware-libs/starknet-specs/pull/77)

I can squash commits if preferred!

After https://github.com/starkware-libs/starknet-specs/pull/84 is merged, we should be able to automatically generate the types of the starknet openrpc API but for now we have to modify them manually.

### RPC version (if applicable)
0.2.1

## Usage related changes

- Introduce expected breaking changes in the starknet.js RPC Provider API

## Development related changes
None

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [ ] Documented the changes
- [ ] Updated the docs (www)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

**Comment by tabaktoni** [comment]
Hello @YohanTz, and thank you for contributing.
What is the status of RPC implementation, we will need to deal with this soon.

**Comment by YohanTz** [comment]
Hey @tabaktoni ! I paused this one as I was waiting for your new pathfinder node so I can test everything :) Is it set up now ?

**Comment by ivpavici** [comment]
I think Infura is running the pathfinder 0.5.3, so it should be able to test over there!

**Comment by YohanTz** [comment]
Ok perfect 👍

File: www/docs/API/Provider/rpcProvider.md [hunk 1]
```diff
@@ -299,12 +299,12 @@ Gets all the events filtered
 
 ```typescript
 type EventFilter = {
-  fromBlock: string;
-  toBlock: string;
-  address: string;
-  keys: string[];
-  page_size: number;
-  page_number: number;
+  fromBlock?: string;
```
[the remark this answers is not in the document]
Comment (YohanTz) [reply www/docs/API/Provider/rpcProvider.md:302]: Oh ok ! Let me remove this

**Pull request closed without merging**